src/cmap/wait_queue.rs

```diff
@@ -55,12 +55,11 @@ impl WaitQueue {
         let future = self.semaphore.acquire(1);
 
         let releaser = if let Some(timeout) = self.timeout {
-            RUNTIME
-                .await_with_timeout(Box::pin(future), timeout)
-                .await
-                .map_err(|_| ErrorKind::WaitQueueTimeoutError {
+            RUNTIME.timeout(timeout, future).await.map_err(|_| {
```

Comment:
now that we use tokio/async-std's builtin timeouts, we don't need the `Box` anymore, which is nice.